src/Api/Controllers/CollectionsController.cs

```diff
@@ -311,6 +306,18 @@ public async Task PostBulkCollectionAccess([FromBody] BulkCollectionAccessReques
             throw new NotFoundException("One or more collections not found.");
         }
 
+        var orgId = collections.First().OrganizationId;
+        if (collections.All(c => c.OrganizationId != orgId))
+        {
+            throw new BadRequestException("All collections must belong to the same organization");
+        }
+
+        // Authorization logic assumes flexible collections is enabled
+        if (!await FlexibleCollectionsIsEnabledAsync(orgId))
+        {
+            throw new NotFoundException();
+        }
```

Comment:
This is basically replicating the `[RequireFeature(FeatureFlagKeys.FlexibleCollections)]` guard, except using the org property.